File: templates/libgdx-game/lwjgl3/src/main/java/com/playroom/runtime/lwjgl3/Lwjgl3Launcher.java
package com.playroom.runtime.lwjgl3;

import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application;
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3ApplicationConfiguration;
import com.playroom.runtime.GameKitGame;
import com.playroom.runtime.services.MockGameServices;

public class Lwjgl3Launcher {
    public static void main(String[] args) {
        Lwjgl3ApplicationConfiguration config = new Lwjgl3ApplicationConfiguration();
        config.setTitle("Playroom Game");
        config.setWindowedMode(390, 844);
        config.useVsync(true);
        config.setForegroundFPS(60);
        new Lwjgl3Application(new GameKitGame(new MockGameServices()), config);
    }
}

File: templates/libgdx-kmp-game/lwjgl3/src/main/kotlin/com/playroom/runtime/lwjgl3/Lwjgl3Launcher.kt
package com.playroom.runtime.lwjgl3

import com.badlogic.gdx.backends.lwjgl3.Lwjgl3Application
import com.badlogic.gdx.backends.lwjgl3.Lwjgl3ApplicationConfiguration
import com.playroom.runtime.GameKitGame

object Lwjgl3Launcher {
    @JvmStatic
    fun main(args: Array<String>) {
        createApplication()
    }

    private fun createApplication(): Lwjgl3Application {
        return Lwjgl3Application(GameKitGame(), defaultConfiguration)
    }

    private val defaultConfiguration: Lwjgl3ApplicationConfiguration
        get() {
            val config = Lwjgl3ApplicationConfiguration()
            config.setTitle("Playroom Game (Kotlin)")
            config.useVsync(true)
            config.setForegroundFPS(60)
            config.setWindowedMode(480, 800)
            return config
        }
}
